Update the policy docs to align with Contributor Covenant 3 (#91)
django/code-of-conduct: working-group-manual.md

- Conference and presentation rooms
- Tutorial and workshop spaces
- Sprint and hackathon areas
- Sponsor and exhibitor halls
- Social events and meal areas
- Hallways, walkways, and common areas that connect event spaces

The Code of Conduct also applies to interactions with official event accounts and spaces on social media and digital platforms, including:

- Comments on official event hashtags
- Posts on event social media accounts
- Event video hosting platforms and live streams
- Official event communication channels (Slack, Discord, etc.)

Django-sponsored events (such as DjangoCon conferences) are required to have their own Code of Conduct and Code of Conduct team. Event organizers will enforce the code throughout the event. Issues at sponsored events can be escalated to the Django Code of Conduct Working Group by:

- The event's Code of Conduct team
- The reporter
- The reported party

The Code of Conduct Working Group will coordinate with event organizers to determine appropriate responses that respect both the event's autonomy and Django's community standards. For detailed guidance on supporting event organizers, see [events.md](events.md).

### Django Online Spaces

> This Code of Conduct applies to the following online spaces managed by the Django project or Django Software Foundation:

Reviewer: Should this section account for communities who opt-into Django's CoC oversight? Specifically Djangonaut Space and Django Commons both note that a person can file a coc violation with the DSF if they are uncomfortable with those that would be reviewing the report in those orgs.

Author: I added that to the end of the list. Thanks!

# Django Code of Conduct - Working Group Manual

This is the internal manual for Django's Code of Conduct Working Group. It documents our procedures, processes, and guidelines for handling Code of Conduct reports and enforcing the Code of Conduct consistently and fairly.

While this is an internal working document, we publish it publicly in the interests of transparency and to help other communities develop their own enforcement processes.

> 

Reviewer 2: ```suggestion

NOTE: Within the scope of this document, any reference to the “working group” shall mean the "Code of Conduct Working Group", unless explicitly specified otherwise.

```

## Table of Contents

- [The Code of Conduct Working Group](#the-code-of-conduct-working-group)
- [Coordination with Online Community Working Group](#coordination-with-online-community-working-group)
- [Scope](#scope)
- [Report Handling Procedures](#report-handling-procedures)
- [Evaluating Reports](#evaluating-reports)
- [Communications Guidelines](#communications-guidelines)
- [Resolutions and Enforcement](#resolutions-and-enforcement)
- [Record Keeping](#record-keeping)
- [Confidentiality](#confidentiality)
- [Conflicts of Interest](#conflicts-of-interest)
- [Acting Unilaterally](#acting-unilaterally)
- [Transparency](#transparency)

## The Code of Conduct Working Group

All responses to reports of conduct violations will be managed by the [Code of Conduct Working Group](https://www.djangoproject.com/foundation/teams/#code-of-conduct-team).

> The Django Software Foundation's Board of Directors ("the board") will establish this working group, comprised of at least three members. One member will be designated chair of the working group and will be responsible for all reports back to the board. The board will review membership on a regular basis.

Reviewer 2: ```suggestion
The [Django Software Foundation's Board of Directors](https://www.djangoproject.com/foundation/#board) ("the board") will establish this working group, comprised of at least three members. One member will be designated chair of the working group and will be responsible for all reports back to the board. The board will review membership on a regular basis.
```

just a nitpick to include a link

> For information about current and past members, see [membership.md](membership.md).

Reviewer 2: ```suggestion
For information about current and past members of the Code of Conduct Working Group, see [membership.md](membership.md).
```

## Coordination with Online Community Working Group

The [Online Community Working Group](https://github.com/django/dsf-working-groups/blob/main/active/online-community.md) handles day-to-day moderation of Django's online spaces and has flexibility to address minor infractions within those spaces using their own moderation processes.

The Code of Conduct Working Group becomes involved when:

- A formal report is submitted directly to the Code of Conduct Working Group

> - A violation occurs across multiple Django spaces

Reviewer 2: ```suggestion
- A violation occurs across multiple Django spaces, which include but are not limited to: Django organization on GitHub, Django Discord, DSF Slack, Django Forum.
```
Should we include the platforms? (I agree they are implicit, anyway)

- The Online Community Working Group escalates an issue that requires formal Code of Conduct enforcement
- An issue at a Django event or sponsored event is escalated to the working group

Each working group designates a liaison from its own members to facilitate coordination and communication between the groups.

## Scope

This Code of Conduct applies to all spaces managed by the Django project or Django Software Foundation. The Code of Conduct also applies when an individual is officially representing the Django community in public spaces.

### Django Events

This Code of Conduct applies to the following people at events hosted or sponsored by the Django Software Foundation:

- DSF staff and board members
- Speakers and panelists
- Tutorial or workshop leaders
- Sprint leaders and mentors
- Organizers and volunteers
- Sponsors and exhibitors
- All attendees

The Code of Conduct applies in official event spaces, including:

- Conference and presentation rooms
- Tutorial and workshop spaces
- Sprint and hackathon areas
- Sponsor and exhibitor halls
- Social events and meal areas
- Hallways, walkways, and common areas that connect event spaces

> 

Reviewer 2: ```suggestion
Any physical spaces within the venue premises where attendees, speakers, sponsors, or staff may interact
```

The Code of Conduct also applies to interactions with official event accounts and spaces on social media and digital platforms, including:

- Comments on official event hashtags
- Posts on event social media accounts
- Event video hosting platforms and live streams
- Official event communication channels (Slack, Discord, etc.)

Django-sponsored events (such as DjangoCon conferences) are required to have their own Code of Conduct and Code of Conduct team. Event organizers will enforce the code throughout the event. Issues at sponsored events can be escalated to the Django Code of Conduct Working Group by:

- The event's Code of Conduct team

> - The reporter

Reviewer 2: ```suggestion
- The reporter: The person/group who submits the complaint
```

> - The reported party

Reviewer 2: ```suggestion
- The reported party: The person/group who is alleged to have violated the Code of Conduct
```

- Mailing lists hosted on djangoproject.com
- GitHub repositories, issue trackers, and pull requests under Django organizations
- Official Django social media accounts
- Any other online space administered by the Django project or Django Software Foundation
- Any other space where their Code of Conduct specifically states that the Django Code of Conduct applies (e.g., [Djangonaut Space](https://github.com/djangonaut-space/program/blob/main/CODE_OF_CONDUCT.md#what-if-your-report-concerns-a-possible-violation-by-an-organizer))

This Code of Conduct applies to the following people in Django online spaces:

- DSF board members and staff
- Django core developers and contributors
- Space administrators and moderators
- All community members and participants

Each online space is encouraged to provide:

- A welcome message with a link to the Code of Conduct
- Clear contact information for reporting violations (<conduct@djangoproject.com>)
- Identification of space moderators or administrators

### Official Representation

The Code of Conduct applies when an individual is officially representing the Django community in public spaces. Examples of official representation include:

> - Using an official Django email address

Reviewer 2: ```suggestion
- Using an official Django email address <@djangoproject.com>
```

  - Inform the working group that the issue is considered resolved.
  - Update the records with a summary of:
    - Actions taken
    - Outcome
    - Review date

The records are kept for reference, are reviewed from time to time, and are consulted regularly to compile statistics. See [Record Keeping](#record-keeping) below for more details.

If the issue was already resolved by local representatives handling a Code of Conduct issue, our working group doesn't carry the responsibility of informing the subject of the report that the record will be kept by us unless there is a clear reason why we would like to follow up to find out their side of the story.

## Evaluating Reports

When reviewing an incident, the working group will evaluate three key aspects:

### Jurisdiction

- Is this behavior explicitly listed in our Code of Conduct as inappropriate?
- Did this occur within a space covered by our Code of Conduct's scope?
- If the incident occurred outside Django spaces but may negatively impact a community member's safety or wellbeing, should it be considered in scope?

### Impact

> - Did this incident occur publicly or privately? Public incidents affect more community members.

Reviewer 2: ```suggestion
- Did this incident occur publicly or privately? Public incidents may affect the broader community. Private incidents, though less visible, may cause substantial harm and are evaluated with equal seriousness.
```

### Communicating About Repair and Accountability

Aligned with Contributor Covenant 3.0's focus on repairing harm, our communications should emphasize accountability, growth, and community healing.

- **Focus on impact, not intent.** When discussing violations with reported parties, center the conversation on the harm caused rather than debating their intentions. Acknowledge that harm can occur even when unintended.

- **Be clear about expectations.** Communicate specific, actionable expectations for behavior change. Vague requests like "be better" are less effective than concrete guidance such as "Do not contact Jin directly; communicate only through the working group."

- **Support accountability.** When appropriate enforcement actions include opportunities for repair (such as apologies or acknowledgment of responsibility), provide guidance on what meaningful accountability looks like. This might include acknowledging the specific harm caused, demonstrating understanding of the impact, and committing to behavior change.

> - **Use trauma-informed language.** Avoid language that minimizes harm ("just a misunderstanding") or blames those who were harmed ("they were too sensitive"). Use person-first language that separates the behavior from the individual's identity.

Reviewer 2: ```suggestion
- **Use trauma-informed language.** Avoid language that minimizes harm ("just a misunderstanding") or blames those who were harmed ("they were too sensitive"). Use person-first language that separates the behavior from the individual's identity. Refer to “the individual whose conduct was reported” rather than labeling someone by the behavior, and to “the individual who experienced harm” rather than defining someone by the harm.
```

Should we include examples of the person-first approach? Adding a few here, if we plan to include them:
> Not Recommended Version: "The offender was rude and made the victim uncomfortable during the workshop."
Recommended Person-First Version: "During the workshop session, the individual whose conduct was reported, interrupted another participant multiple times while they were speaking.
The individual who reported the incident stated that they felt uncomfortable and discouraged from continuing their participation."

3. **Report Documentation:** Individual Google Docs for each case.

All documents are shared only between members of the working group. The working group insists on not sharing the records outside the members due to confidentiality of information. Violating this will result in removal of the member from the working group.

**Privacy Considerations:**

- Always share documents with specific working group members rather than turning link sharing on
- Use code names in document titles instead of real names or identifiers - even if a person doesn't have edit or view access to an individual report, they can still see document titles in shared folders
- Code names prevent inadvertent identification and make it easier to discuss cases in public spaces (such as events)
- **Identity Key Access:** Access to the Person Identity Key may be restricted to a smaller subset of the working group if necessary to manage conflicts of interest.

**Member Access:**

- Once a member steps down from the working group, their access to the records is removed
- When onboarding new working group members, they should be provided with a list of code names for all existing reports

> - The new member should state whether they have any conflicts of interest with reviewing documentation for those cases

Reviewer 2: ```suggestion
- The new member should state whether they have any conflicts of interest with reviewing documentation for those cases. Refer [Conflicts of Interest](#conflicts-of-interest).
```

If the incident involves physical danger, any member of the working group must act unilaterally to protect safety. This can include contacting law enforcement (or other local personnel) and speaking on behalf of the DSF.

In situations where an individual working group member acts unilaterally, they must report their actions to the working group for review within 24 hours.

## Transparency

The Code of Conduct working group is committed to transparency about our processes, decisions, and activities. For complete information about our transparency practices, including what we disclose and what we protect, see [Transparency](transparency.md).

Key transparency practices include:

- **Publishing this manual and other documentation** about our procedures
- **Publishing statistics** every 6 months (see [statistics.md](statistics.md))
- **Annual transparency reports** with anonymized data and trends
- **Never publicly discussing specific incidents** - all public statements are made by the DSF Board if necessary

We balance transparency with our commitment to protecting the confidentiality of reporters, harmed individuals, and reported parties.

> 

Reviewer 2: Two open points:

- Do we need guidance on handling knowingly false or bad-faith reports or spam, per se?
- Do we need documentation on if the Online Community WG and CoC WG disagree on escalation or outcome, how to reach a common ground?

Author: Both great questions. I've gone back and forth with bad-faith reporting because we can't know a person's thoughts or motivations, only their actions. I think spam or repeatedly reporting the same issue when we've already decided that issue isn't a violation is something we can define though. I've added language to that affect to the new CoC at https://github.com/django/code-of-conduct/blob/docs/adopt-contributor-covenant/CODE_OF_CONDUCT.md#other-restrictions.

Great point about a misalignment between the two WGs. I'll add language around that.
